UserModel.js

```diff
@@ -0,0 +1,37 @@
+const mongoose = require('mongoose');
+const bcrypt = require('bcrypt');
+
+const UserSchema = new mongoose.Schema({
+    username: {
+        type: String,
+        required: true,
+        unique: true,
+        lowercase: true,
+    },
+    password: {
+        type: String,
+        minlength: 6,
+        required: true
+    }
+});
+
+//hash functionality using bcrypt
+UserSchema.pre('save', function(next) {
```

Comment:
Solid use of Schema for Validation and Encryption with pre save hooks and methods.